src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -173,10 +173,14 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            // restart the timer with the next schedule occurrence
             await InvokeJobFunction(DateTime.Now, false);
 
-            StartTimer(DateTime.Now);
+            // Restart the timer with the next schedule occurrence, but only 
+            // if Cancel, Stop, and Dispose have not been called.
+            if (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                StartTimer(DateTime.Now);
```

Comment:
We need to use Local time not UTC time for TimerTrigger, to allow people to control their time zone in their Web App. See https://github.com/Azure/azure-webjobs-sdk-extensions/issues/32 for all the details :) We used to use UTC then fixed the issues by moving to Local.